addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CrossSiteScriptingScanRule.java

```diff
@@ -677,6 +678,81 @@ private boolean performImageTagAttack(HtmlContext context, HttpMessage msg, Stri
         return false;
     }
 
+    private boolean performAttributeAttack(HtmlContext context, HttpMessage msg, String param) {
+        String attackString = "accesskey=\"x\" onclick=\"alert('1')\" b";
+        List<HtmlContext> context2 = performAttack(msg, param, attackString, context, 0);
+        if (context2 == null) {
+            return false;
+        }
+        for (HtmlContext ctx : context2) {
+            HashMap<String, String> attrMap = ctx.getTagAttributes();
+            if (attrMap.containsKey("accesskey")
+                    && attrMap.get("accesskey").equals("x")
+                    && attrMap.containsKey("onclick")
+                    && attrMap.get("onclick").equals("alert('1')")
+                    && attrMap.containsKey("b")
+                    && attrMap.get("b").equals("")) {
+                // Yep, its vulnerable
+                newAlert()
+                        .setConfidence(Alert.CONFIDENCE_MEDIUM)
+                        .setParam(param)
+                        .setAttack(ctx.getTarget())
+                        .setEvidence(ctx.getTarget())
+                        .setMessage(ctx.getMsg())
+                        .raise();
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private boolean performElementAttack(HtmlContext context, HttpMessage msg, String param) {
+        String attackString1 = "tag accesskey=\"x\" onclick=\"alert('XSSED')\"/";
```

Comment:
`alert(1)` again. As well as avoiding any code checking for single quotes it also makes the attack string slightly shorter which could make the difference in some cases :)